classes/Customer.php

```diff
@@ -1221,17 +1221,20 @@ public function transformToCustomer($idLang, $password = null)
 
         /*
         * If this is an anonymous conversion and we want the customer to set his own password,
-        * we set a random one for now. If a password was provided, we check it's validity.
+        * we set a random one for now.
         */
         if (empty($password)) {
-            $this->passwd = $crypto->hash(Tools::passwdGen(16, 'RANDOM'));
+            $password = Tools::passwdGen(16, 'RANDOM');
         } else {
             if (!Validate::isAcceptablePasswordLength($password) || !Validate::isAcceptablePasswordScore($password)) {
                 return false;
             }
-            $this->passwd = $crypto->hash($password);
         }
 
+        /** @var \PrestaShop\PrestaShop\Core\Crypto\Hashing $crypto */
```

Comment:
Please don't merge anything in this method, it's already fixed on develop branch.